### PR TITLE
Admin user index

### LINKED ISSUE #53: User Story 53, Admin User Index Page
As an admin user
When I click the "Users" link in the nav (only visible to admins)
Then my current URI route is "/admin/users"
Only admin users can reach this path.
I see all users in the system
Each user's name is a link to a show page for that user ("/admin/users/5")
Next to each user's name is the date they registered
Next to each user's name I see what type of user they are

### PR DESCRIPTION
**Pull request by c-bartell**
Adds an admin-accessible users index page.
---
- Updates "All Users" link to "Users" in the admin nav bar
- Adds a users index page which lists all users, their role, and registration date
- Each user's name is a link to their admin users profile page
- All tests passing
- Closes #53

### CODE REVIEW

**Review by JoseLopez235 — APPROVED** [review]

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Admin::UsersController < Admin::BaseController
+  def index
```
Comment (stillsheryl) [inline app/controllers/admin/users_controller.rb:2]: Nice idea to use the admin users controller here

File: spec/features/admin/users/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+              expect(page).to have_content("Permissions: #{@admin.role.upcase}")
+              expect(page).to have_content("Registration Date: #{@admin.created_at.to_date}")
+
+              click_link(@admin.name)
+          end
+        end
+
+        expect(current_path).to eq("/admin/users/#{@admin.id}")
+
+        visit admin_users_path
```
Comment (stillsheryl) [inline spec/features/admin/users/index_spec.rb]: Just curious, what's this doing?

File: spec/features/admin/users/index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+      click_button 'Login'
+
+      within "nav" do
+        click_link('Users')
+      end
+    end
+
+    describe "When I click on the 'Users' link" do
+      it "my current URL route is '/admin/users'" do
+        expect(current_path).to eq(admin_users_path)
```
Comment (stillsheryl) [inline spec/features/admin/users/index_spec.rb:27]: Nice job using the path helpers!

**Review by stillsheryl — APPROVED** [review]

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -20,7 +20,7 @@
         <%= content_tag :p, "Logged in as #{current_user.name}" %>
       <% elsif current_admin? %>
         <%= link_to "Admin's Dashboard", "/admin" %>
-        <%= link_to "All Users", "/admin/users" %>
+        <%= link_to "Users", "/admin/users" %>
```
Comment (KieraAllen) [inline app/views/layouts/application.html.erb:23]: I have been awaiting this moment (where we have a page to see all users) for ages! I'm excited we can do this now (as an admin user)

**Review by KieraAllen — APPROVED** [review]

File: spec/features/admin/users/index_spec.rb (hunk 2 above)
Comment (c-bartell) [reply spec/features/admin/users/index_spec.rb]: It's not doing anything 😅. I updated the PR to take it out.

**Pull request merged**